crates/dao/src/value.rs

```diff
@@ -331,7 +331,7 @@ fn parse_naive_date_time(v: &str) -> NaiveDateTime {
     if let Ok(ts) = ts {
         ts
     } else {
-        let ts = NaiveDateTime::parse_from_str(&v, "%Y-%m-%d %H:%M:%S.%f");
```

Comment:
How about the case where there is only 1, 2 or 4  decimal in used for the milliseconds.

Reply:
According to [chrono strftime](https://docs.rs/chrono/0.4.19/chrono/format/strftime/index.html) %f is for parsing nanoseconds, which is not applicable for this format since it carries (AFAIK) only milliseconds. Milliseconds are getting parsed by either %.3f (which prints leading dot) or %3f (which omits printing the leading dot).

My application did write e.g. "2021-03-04T16:45:22.254" to the database, but without this fix, "2021-03-04T16:45:22.000000254" was read back.

So, handling 1 and 2 decimal was ok in my tests, 4 decimals would be microseconds and therefore beyond accuracy milliseconds. I did not test that case, but i assume it would be rounded to milliseconds.

Reply:
Please add the testcase for this.

Reply:
I've added the updated example i was using to test.